app/database/operator/server_data_operator/handlers/channel.ts

```diff
@@ -353,6 +356,87 @@ const ChannelHandler = <TBase extends Constructor<ServerDataOperatorBase>>(super
             tableName: CHANNEL_MEMBERSHIP,
         }, 'handleChannelMembership');
     };
+
+    /**
+     * handleChannelMembership: Handler responsible for the Create/Update/Delete operations occurring on the CHANNEL_BOOKMARK table from the 'Server' schema
+     * @param {HandleChannelBookmarkArgs} channelBookmarkArgs
+     * @param {ChannelBookmark[]} channelBookmarkArgs.bookmarks
+     * @param {boolean} channelBookmarkArgs.prepareRecordsOnly
+     * @returns {Promise<ChannelBookmarkModel[]>}
+     */
+    handleChannelBookmark = async ({bookmarks, prepareRecordsOnly}: HandleChannelBookmarkArgs): Promise<Model[]> => {
+        if (!bookmarks?.length) {
+            logWarning(
+                'An empty or undefined "bookmarks" array has been passed to the handleChannelBookmark method',
+            );
+            return [];
+        }
+
+        const uniqueRaws = getUniqueRawsBy({raws: bookmarks, key: 'id'}) as ChannelBookmarkWithFileInfo[];
+        const keys = uniqueRaws.map((c) => c.id);
+        const db: Database = this.database;
+        const existing = await db.get<ChannelBookmarkModel>(CHANNEL_BOOKMARK).query(
+            Q.where('id', Q.oneOf(keys)),
+        ).fetch();
+        const bookmarkMap = new Map<string, ChannelBookmarkModel>(existing.map((b) => [b.id, b]));
+        const files: FileInfo[] = [];
+        const raws = uniqueRaws.reduce<{createOrUpdateRaws: ChannelBookmarkWithFileInfo[]; deleteRaws: ChannelBookmarkWithFileInfo[]}>((res, b) => {
+            const e = bookmarkMap.get(b.id);
+            if (!e) {
+                if (!b.delete_at) {
+                    res.createOrUpdateRaws.push(b);
+                    if (b.file) {
+                        files.push(b.file);
+                    }
+                }
+                return res;
+            }
+
+            if (e.updateAt !== b.update_at) {
```

Comment:
I don't see the point

Reply:
Let's say... we update this twice in a short amount of time, so we have:
- a.updateAt = 0
- b.updateAt = 1
- c.updateAt = 2

If we, for any reason, process this in the wrong order (we have a, process c, then b) we end up with b, when it should be c.

If we check updateAt < updateAt, even if we process c before b, we end up with c.

Nevertheless, this is all theoretical. 0/5 on how often this may happen.